backend/functions/src/middleware/validate-id-token.ts

```diff
@@ -0,0 +1,16 @@
+import { NextFunction, Request, Response } from 'express';
+import { getAuth } from 'firebase-admin/auth';
+
+export const validateIdToken = (req: Request, res: Response, next: NextFunction) => {
+  if (req.headers.authorization) {
+    getAuth()
+      .verifyIdToken(req.headers.authorization.split(' ')[1])
+      .then(decodedToken => {
+        req.body.user = decodedToken;
+        next();
+      })
+      .catch(error => {
+        next(error);
+      });
+  } else next(new Error("'User not authenticated'"));
```

Comment:
Rewrite this using `async`/`await` with `try`/`catch` block. Also use the `RequestHandler` type annotation on the function. Also, make the error message here more descriptive.